roles/vgpu/tasks/install.yml

```
@@ -0,0 +1,86 @@
+---
+- name: Install OS dependencies
+  vars:
+    pkgs:
+      - unzip
+      - gcc
+      - make
+      - "{{ 'kernel-devel' if ansible_facts.os_family == 'RedHat' }}"
+      # sriov-manage requires lspci
+      - pciutils
+      - mdevctl
+  package:
+    name: "{{ pkgs | select }}"
+    state: present
+  become: true
+
+- block:
+    - name: Ensure target directory exists
+      file:
+        state: directory
+        path: "{{ dir_path }}"
+        owner: root
+        group: root
+        mode: "770"
+
+    - name: Copy driver onto target host
+      copy:
+        src: "{{ vgpu_driver_path }}"
+        dest: "{{ dir_path }}/{{ filename }}"
+        owner: root
+        group: root
+        mode: "660"
+      when: false
+
+    - name: Extract driver
+      unarchive:
+        src: "{{ vgpu_driver_path }}"
+        dest: "{{ dir_path }}"
+        owner: root
+        group: root
+        creates: "{{ dir_path }}/Host_Drivers"
+
+    - name: Find .run script
+      find:
+        paths: "{{ dir_path }}/Host_Drivers"
+        patterns: "*.run"
+      register: find_result
+
+    - name: Set execute bit
+      file:
+        path: "{{ install_script }}"
+        mode: "u+x"
+
+    - name: Run the install script
+      # NOTE: This compiles for currently running kernel, can force with --kernel-name
+      shell: |-
+        {{ install_script }} -q {% if vgpu_driver_dkms %}--dkms{% endif %} --ui none --disable-nouveau --no-nouveau-check && touch {{ install_script }}.complete
+      args:
+        creates: "{{ omit if vgpu_driver_force_install else install_script ~ '.complete' }}"
+      register: install_result
+
+    - name: Reboot after driver install
+      reboot:
+        reboot_timeout: 10000
```

Should we make this shorter? 1800 maybe? (30 minutes)

---

Good plan
```suggestion
        reboot_timeout: 1800
```
Good plan. That was a bit long!